dataans/src-tauri/src/dataans/command/import/json.rs

```diff
@@ -0,0 +1,45 @@
+use crate::dataans::db::Db;
+use crate::dataans::service::note::NoteService;
+use crate::dataans::service::space::SpaceService;
+use crate::dataans::DataansError;
+use common::export::{Schema, SchemaV1};
+use serde_json;
+use std::fs::File;
+use std::path::Path;
+
+pub async fn import_v1<D: Db>(
+    schema_v1: SchemaV1,
+    space_service: &SpaceService<D>,
+    note_service: &NoteService<D>,
+) -> Result<(), DataansError> {
+    for space_data in schema_v1.data {
+        let space = space_data.space;
+        let notes = space_data.notes;
+
+        let space_exists = space_service.space_by_id(space.id).await.is_ok();
+        if !space_exists {
+            space_service.create_space(space).await?;
+        }
+
+        for note in notes {
+            let note_exists = note_service.note_by_id(note.id).await.is_ok();
+            if !note_exists {
+                note_service.create_note(note).await?;
+            }
+        }
+    }
```

Comment:
it's better to run all these futures in parallel. your current implementation adds spaces and notes one by one and doesn't use the benefits of async.

```rust
futures::join_all(schema_v1.data.into_iter().map(|space_data| {
    let space = space_data.space;
    let notes = space_data.notes;
    
    /* *** */
})).await
```

you can also apply the same optimization for the inner loop too